### PR TITLE
User Authentication

### PR DESCRIPTION
**Pull request by danielcooperxyz**
Authentication was previously done using machine to machine authentication. These changes enable users to acces the application and authenticate themselves against Auth0, which is then used for token creation against the blockchain API service.

These changes are only for the basic authentication flow to Auth0 to permit access to the web application and API.

### Customer
- [x] Login flow
- [x] Login styling
- [x] Logout button

### T1
- [x] Login flow
- [x] Login styling
- [x] Logout button

Screenshot 2022-02-28 at 16 43 32
Screenshot 2022-02-28 at 16 38 31
Screenshot 2022-02-28 at 14 27 58
Screenshot 2022-02-28 at 14 23 27

### CODE REVIEW

File: src/customer/components/Navigation.js [hunk 1]
```diff
@@ -102,6 +109,18 @@ const Navigation = () => {
       <Box className={classes.navButton}>
         <NetworkStatusIndicator />
       </Box>
+      {isAuthenticated ? (
```
Comment (darrylmorton-digicatapult) [inline src/customer/components/Navigation.js:112]: isAuthenticated &&

File: src/customer/router/index.js [hunk 2]
```diff
@@ -1,18 +1,46 @@
...
+import { useAuth0 } from '@auth0/auth0-react'
 
 const Routing = () => {
+  const { isAuthenticated, isLoading, loginWithRedirect, user } = useAuth0()
+
+  useEffect(() => {
+    if (!isAuthenticated && !isLoading && !user) {
+      loginWithRedirect()
+    } else {
+      console.log('Authenticated!')
```
Comment (darrylmorton-digicatapult) [inline src/customer/router/index.js:17]: remove console.log

**Review by darrylmorton-digicatapult — CHANGES_REQUESTED** [review]

File: src/customer/router/index.js (hunk 2 above)
Comment (danielcooperxyz) [reply src/customer/router/index.js]: removed, thank you

File: src/customer/components/Navigation.js (hunk 1 above)
Comment (danielcooperxyz) [reply src/customer/components/Navigation.js]: updated!

**Review by n3op2 — DISMISSED** [review]

File: src/utils/url.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+export const getCurrentBaseUrl = () => {
+  const protocol = window.location.protocol
```
[the remark this answers is not in the document]
Comment (danielcooperxyz) [reply src/utils/url.js:2]: nice, thanks

File: src/utils/vitalamApi.js [hunk 4]
```diff
@@ -49,46 +48,16 @@ const useFetchWrapper = () => {
...
-        }
-      )
-
-      token = response.access_token
-      localStorage.setItem('token', token)
-    }
-    return token
+    return await getAccessTokenSilently({
+      audience: AUTH_AUDIENCE,
+      scope: '',
```
[the remark this answers is not in the document]
Comment (danielcooperxyz) [reply src/utils/vitalamApi.js]: have removed, but i had put it there for explicitness as scopes are part of users permissions, at the moment we do not use them, but don't think we ever will

**Review by darrylmorton-digicatapult — APPROVED** [review]

**Review by n3op2 — APPROVED** [review]

**Review by n3op2 — APPROVED** [review]

**Review by n3op2 — APPROVED** [review]

**Review by darrylmorton-digicatapult — APPROVED** [review]

**Review by jonmattgray — APPROVED** [review]

**Pull request merged**